hardware/smart_utils.py

```diff
@@ -19,6 +19,16 @@
 from hardware.detect_utils import which
 
 
+def _parse_line(line):
+    line = line.strip()
+    if isinstance(line, six.binary_type):
+        if line.startswith('Add. Product Id:'):
+            # Remove non-ascii characters from line
+            line = ''.join(i for i in line if ord(i) < 128)
+            line = line.decode()
```

Comment:
this looks like too much indentation

Reply:
I can look into moving this function into detect_utils, as well as adding unit tests. Additionally, there are a number of identical calls to subprocess that we'll likely want to move there as well?

`sdparm_cmd = subprocess.Popen("smartctl -a %s %s" % (device, optional_flag), shell=True, stdout=subprocess.PIPE)`

As far as the indentation, can you point out where? I ran that function individually through pycodestyle and pylint to help me see where the indentation issue is, but they returned without error.

Reply:
L28 should be at the same level as L25, we want to decode all the lines, not only if they start with 'Add. Product Id:'
That's why I asked for a unit test also :)
For now let's keep this commit only about this function, it will be easier to merge, we can work on other similar call in follow ups.

Reply:
Good eyes on the level issue, even if it didn't make it to the final commit! Definitely missed that. 

I'll meet you half way on your request here. As far as re-factoring the code to move the functionality to detect_utils, I believe that should be a separate pull request/action by a code maintainer. The current change is tantamount to moving repeated functionality to it's own function, not adding any additional logic to the script. 

On the other hand, I've added some additional testing to this PR by your suggestion, which includes sample data, and a subsequent, successful decoding data set to compare against. I had to modify the function reading the sample file to prevent it's own reading issues with the sample data. 

Let me know if this compromise is acceptable to you, and we can go from there.

Reply:
The important part was the test, thanks for adding that!
It's totally fine as it is for me :)